dali/operators/image/resize/resize_op_impl_cpu.h

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_OPERATORS_IMAGE_RESIZE_RESIZE_OP_IMPL_CPU_H_
+#define DALI_OPERATORS_IMAGE_RESIZE_RESIZE_OP_IMPL_CPU_H_
+
+#ifndef DALI_RESIZE_BASE_CC
+#error This file is a part of resize base implementation and should not be included elsewhere
+#endif
+
+#include <cassert>
+#include <vector>
+#include "dali/operators/image/resize/resize_op_impl.h"
+#include "dali/kernels/imgproc/resample_cpu.h"
+
+namespace dali {
+
+template <typename Out, typename In, int spatial_ndim>
+class ResizeOpImplCPU : public ResizeBase<CPUBackend>::Impl {
+ public:
+  explicit ResizeOpImplCPU(int num_threads) {
+    kmgr_.Resize(num_threads, 0);
+  }
+
+  static_assert(spatial_ndim == 2, "NOT IMPLEMENTED. Only 2D resize is supported");
+
+  using Kernel = kernels::ResampleCPU<Out, In, spatial_ndim>;
+
+  /// Dimensnionality of each separate frame. If input contains no channel dimension, one is added
+  static constexpr int frame_ndim = spatial_ndim + 1;
+
+  void Setup(TensorListShape<> &out_shape,
+             const TensorListShape<> &in_shape,
+             int first_spatial_dim,
+             span<const kernels::ResamplingParams> params) override {
+    GetResizedShape(out_shape, in_shape, params, spatial_ndim, first_spatial_dim);
+
+    // Create "frames" from outer dimensions and "channels" from inner dimensions.
+    GetFrameShapesAndParams<spatial_ndim>(in_shape_, params_, in_shape, params,
+                                          first_spatial_dim);
+
+    // Now that we have per-frame parameters, we can calculate the output frame shape.
+    GetResizedShape(out_shape_, in_shape_, make_cspan(params_), 0);
```

Comment:
It still a bit confusing seeing to calls to just a different variant of GetResizedShape, separated by a GetFrameShapesAndParams call. It looks like: do something, rest, try harder.

Reply:
The first one calculates `out_shape`, the other `out_shape_`.

Reply:
Still reading this code confuses me.

Reply:
I'll be more elaborate in the comments.